mlir/lib/Interfaces/ValueBoundsOpInterface.cpp

```diff
@@ -107,25 +107,57 @@ AffineExpr ValueBoundsConstraintSet::getExpr(Value value,
   assertValidValueDim(value, dim);
 #endif // NDEBUG
 
+  // Helper function that returns an affine expression that represents column
+  // `pos` in the constraint set.
+  auto getPosExpr = [&](int64_t pos) {
+    assert(pos >= 0 && pos < cstr.getNumDimAndSymbolVars() &&
+           "invalid position");
+    return pos < cstr.getNumDimVars()
```

Comment:
This ternary condition feels like an implementation detail that appears everywhere.
Can we give it a proper name and hide away the impl ?

Reply:
I have a WIP change that turns this into a member function (so that it can be called from various places). Can also be done now, I moved it into this PR.